Improve wording of AttachAsync guidance

Diego-Perez-Botero opened the pull request:
Mention of "events" is confusing. Also, UploadOperations are subject to the same guidance. Most importantly, AttachAsync is something that should happen during app launch - it doesn't make sense to do it every time that a transfer is being started (like the current docs seem to suggest).

Diego-Perez-Botero committed:
Improve wording of AttachAsync guidance

Mention of "events" doesn't make any sense. Also, UploadOperations are subject to the same guidance.

mikematteson commented:
@stevewhims : PR **passed** acceptance checklist. Please review further for technical accuracy, language, etc. Thanks!

stevewhims committed:
Just a slight tweak

The feature itself won't be rendered useless; only the customer's application/use/realization of that feature.

stevewhims reviewed (approved):
Looks great; I just made one philosophical tweak. :) Thanks, Diego!
